geospaas/base_viewer/forms.py

```diff
@@ -0,0 +1,74 @@
+from django.contrib.gis import forms
+from django.db.models import Q
+from django.utils import timezone
+from leaflet.forms.widgets import LeafletWidget
+
+from geospaas.base_viewer.models import SearchModelForLeaflet
+from geospaas.catalog.models import Dataset as CatalogDataset
+
+#from geospaas.base_viewer.models import Search_with_para
+
+class BaseForm(forms.ModelForm):
```

Comment:
I am having trouble to understand why you use `ModelForm` as the base class.
From what I understand, `ModelForm` is used when you want to create/update instances of Models in your database.
You only have search forms here, so you can probably just use the `Form` class as base.

Reply:
django `ModelForm` is a 'direct bridge' of our models. These kind of forms are specified based on the specifications of columns of our database. Thus they are easily set and used and the user can correctly behave with them since they are individually customized with individuals of our columns of database. This is an ability of Django which make things easy for us to use this kind of form and then stack them in a 'html form' for further processing. In this way, we can do what we want in an `form agnostic` way in `views.py`. `form agnostic` by our mean is that `do the logic in views regardless of type of form and type of data that we work with it`. This might not possible without django `ModelForm`(or maybe extreme difficult and needs lots of lines of code to be operational).

Reply:
Actually, you don't use much of the functionalities of `ModelForm`, so I am pretty sure you can just replace your forms by the following:
```
class TimeAndSourceForm(forms.Form):
    time_coverage_start = forms.DateTimeField(initial=timezone.datetime(2000, 1, 1))
    time_coverage_end = forms.DateTimeField(initial=timezone.now())
    source = forms.ModelChoiceField(Source.objects.all())

    def filter(self, ds):
        [...]

class SpatialSearchForm(forms.Form):
    polygon = forms.GeometryField(widget=LeafletWidget(attrs={
        'settings_overrides': {
            'DEFAULT_CENTER': (60.0, 5.0),
            'DEFAULT_ZOOM': 1,
            'PLUGINS': {'forms': {'auto-include': True}},
        }
    }))

    def filter(self, ds):
        [...]
```

This also removes the need for the `BaseForm` class and the `SearchModelForLeaflet` model.

Reply:
YES, you have also a corrcet approach. But when django automate things for us, why don't we take advantage of it? All of those `DateTimeField`, and`ModelChoiceField` explicit definition of fields can be done by django implicitly(automatically). This way we can inherit from this form in the ADASviewer and remove each of these fields when we want them to remove. Modelform provides more flexibility for us to **remove** or **add** fields in possible inheritance in future.

I discussed it previously with @akorosov. I had your opinion previously,but now I have found that this could be the better way of implementing it.